### PR TITLE
fix: use noop insights by default on maestro

### PR DESCRIPTION
**Pull request by amanjeetsingh150**
## Proposed changes

`Insights` is a CLI-only concept. We've seen this implementation causing problems on the cloud. This PR provides a noop implementation to Insights by default so that we don't use it on cloud.

For CLI the real implementation is passed so that behavior of CLI remains unchanged

## Testing

Locally running a command that raises insight. Tapping on optional view that's not there.

### CODE REVIEW

**Review by axelniklasson — APPROVED** [review]

**Review by igorsmotto — APPROVED** [review]

File: maestro-utils/src/main/kotlin/Insight.kt [hunk 1]
```diff
@@ -1,31 +1,20 @@
 package maestro.utils
 
-object Insights {
+object Insights: InsightsInterface {
```
Comment (herval) [inline maestro-utils/src/main/kotlin/Insight.kt:3]: IIUC, this doesn't do anything, other than invoking a bunch of listeners. Why do you need a noop version in that case?

File: maestro-utils/src/main/kotlin/InsightsInterface.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package maestro.utils
+
+interface InsightsInterface {
```
Comment (herval) [inline maestro-utils/src/main/kotlin/InsightsInterface.kt:3]: nit: interfaces shouldn't include "Interface" on their name.

File: maestro-utils/src/main/kotlin/Insight.kt (hunk 1 above)
Comment (amanjeetsingh150) [reply maestro-utils/src/main/kotlin/Insight.kt:3]: The Insights are only used for CLI, the noop is default implementation that we pass for cloud versions and only override for CLI. Earlier the implementations were coupled inside Orchestra and hence the cloud code.

File: maestro-utils/src/main/kotlin/InsightsInterface.kt (hunk 2 above)
Comment (amanjeetsingh150) [reply maestro-utils/src/main/kotlin/InsightsInterface.kt:3]: Good point, I'll fix this

Comment (amanjeetsingh150) [reply maestro-utils/src/main/kotlin/InsightsInterface.kt]: Done ✅

**Pull request merged**